pyccel/codegen/make_pipeline.py

```diff
@@ -0,0 +1,347 @@
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/devel/LICENSE for full license details.      #
+#------------------------------------------------------------------------------------------#
+"""
+Contains the execute_pyccel function which carries out the main steps required to execute pyccel
+"""
+
+import os
+import shutil
+import sys
+import time
+from pathlib import Path
+
+from pyccel.errors.errors          import Errors, PyccelError
+from pyccel.errors.errors          import PyccelSyntaxError, PyccelSemanticError, PyccelCodegenError
+from pyccel.errors.messages        import PYCCEL_RESTRICTION_TODO
+from pyccel.parser.parser          import Parser
+from pyccel.codegen.build_generation.cmake_gen import CMakeHandler
+from pyccel.codegen.build_generation.meson_gen import MesonHandler
+from pyccel.codegen.codegen        import Codegen
+from pyccel.codegen.compiling.project import CompileTarget, BuildProject
+from pyccel.codegen.utilities      import manage_dependencies
+from pyccel.codegen.wrappergen     import Wrappergen
+from pyccel.naming                 import name_clash_checkers
+from pyccel.utilities.stage        import PyccelStage
+from pyccel.parser.scope           import Scope
+
+from .compiling.compilers import Compiler, get_condaless_search_path
+
+pyccel_stage = PyccelStage()
+
+__all__ = ['execute_pyccel_make']
+
+build_system_handler = {'cmake': CMakeHandler,
+                        'meson': MesonHandler}
+
+#==============================================================================
+# NOTE:
+# [..]_dirname is the name of a directory
+# [..]_dirpath is the full (absolute) path of a directory
+
+def execute_pyccel_make(files, *,
+                   verbose,
+                   time_execution,
+                   folder,
+                   language,
+                   compiler_family,
+                   build_system,
+                   debug = None,
+                   accelerators,
+                   conda_warnings,
+                   build_code):
+    """
+    Run Pyccel-make on the provided files.
+
+    Carry out the main steps required to execute Pyccel:
+    - Parses the python file (syntactic stage)
+    - Annotates the abstract syntax tree (semantic stage)
+    - Generates the translated file(s) (codegen stage)
+    - Generates the wrapper file(s) (wrapper stage)
+    - Generates the build system file(s)
+    - Compiles the files to generate executable(s) and/or shared library(s).
+
+    Parameters
+    ----------
+    files : list[Path]
+        The Python files to be translated.
+    verbose : int
+        Indicates the level of verbosity.
+    time_execution : bool
+        Show the time spent in each of Pyccel's internal stages.
+    folder : str
+        Path to the working directory. Default is the folder containing the file to be translated.
```

Comment:
Is this the directory where the output files should be generated (which may vary in a multi-file project), or the directory where the `__pyccel__` folder should be created?